### PR TITLE
Module split PR3

### PR DESCRIPTION
**Pull request by Pessimistress**
#### Background
Module split for #1629

#### Change List
- Config workspaces for yarn
- Remove unused fields from root package.json
- Add bootstrap and publish scripts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "description": "A suite of 3D-enabled data visualization overlays, suitable for react-map-gl",
   "license": "MIT",
   "version": "5.2.0-alpha.4",
```
Comment (jianhuang01) [inline package.json]: we are in alpha.8 now.

File: package.json [hunk 2]
```diff
@@ -13,28 +14,19 @@
...
+    "clean": "rm -r node_modules && lerna clean --yes",
     "cover": "NODE_ENV=test nyc --reporter html --reporter cobertura --reporter=lcov npm run test-cover",
-    "pretty": "prettier --write \"{src,examples,test}/**/*.js\" --loglevel warn",
-    "lint": "prettier-check \"{src,examples,test}/**/*.js\" && eslint src test examples && npm run lint-yarn",
+    "pretty": "prettier --write \"{modules,examples,test}/**/*.js\" --loglevel warn",
+    "lint": "prettier-check \"{modules,examples,test}/**/*.js\" && eslint modules test examples && npm run lint-yarn",
     "lint-yarn": "!(grep -q unpm.u yarn.lock) && echo 'Lockfile valid.' || (echo 'Please rebuild yarn file using public npmrc' && false)",
-    "publish-prod": "npm run build && npm run test && npm run test-dist && npm publish",
-    "publish-beta": "npm run build && npm run test && npm run test-dist && npm publish --tag beta",
+    "publish-prod": "learna run build && npm run test && npm run test-dist && scripts/publish.sh prod",
```
Comment (ibgreen) [inline package.json]: Should that be `lerna`?

File: package.json [hunk 3]
```diff
@@ -83,6 +63,7 @@
     "gl": "^4.0.3",
     "html-webpack-plugin": "^3.1.0",
     "jsdom": "^9.11.0",
+    "lerna": "^2.9.1",
     "module-alias": "^2.0.0",
     "nyc": "^10.2.0",
     "pixelmatch": "^4.0.2",
```
Comment (ibgreen) [inline package.json]: Should pixelmatch come from `test-utils`?

File: scripts/bootstrap.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+# Script to bootstrap repo for development
+
+set -e
+
+# install dependencies
+yarn
+
+ROOT_MODULES=`pwd`/node_modules
```
Comment (ibgreen) [inline scripts/bootstrap.sh]: Nit: Rename to `ROOT_NODE_MODULES_DIR` to distinguish from other `modules` being handled in this file.

File: scripts/publish.sh [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+ACCOUNT=`npm whoami`
+
+if [[ $ACCOUNT != 'deck.gl' ]]; then
+  echo "Must sign in to deck.gl account to publish"
+  exit 1
+fi
+
+case $MODE in
+  "beta")
+    lerna publish --npm-tag beta --cd-version prerelease
```
Comment (ibgreen) [inline scripts/publish.sh]: May add short comments about what the `lerna` options do?

File: scripts/bootstrap.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+cd modules
+for D in *; do (
+  [ -d $D ]
+  cd $D
+
+  # create symlink to dev dependenciess at root
```
Comment (ibgreen) [inline scripts/bootstrap.sh]: typo `ss`

File: scripts/bootstrap.sh [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  # this is a bug of yarn: https://github.com/yarnpkg/yarn/issues/4964
+  # TODO - remove when fixed
+  mkdir -p node_modules
+  ln -sf $ROOT_MODULES/.bin ./node_modules
+
+  # build the module
+  npm run prepublish
```
Comment (ibgreen) [inline scripts/bootstrap.sh]: See concerns about `prepublish` from previous diffs.

**Pull request merged**